### ISSUE TITLE
Mixins with & selectors make sourcemaps invalid

### ISSUE THREAD

**Issue by chaance** [sass/libsass#2617]
Looks a lot like https://github.com/sass/libsass/issues/2312, perhaps not completely resolved. Also seen in https://github.com/sass/libsass/issues/2515

## input.scss

```scss
// Top content boxes.
@mixin top-content {
	background-color: $dark-blue;
	color: #fff;
	padding: 15px;
	border-radius: 4px;

	&__container { // This is the error line

		@include squeeze;
	}
}
```

## Actual results

```
/Users/cs/Desktop/local/pya/app/public/wp-content/themes/pyapc/node_modules/webpack-sources/node_modules/source-map/lib/source-map-generator.js:298
      throw new Error('Invalid mapping: ' + JSON.stringify({
      ^

Error: Invalid mapping: {"generated":{"line":11932,"column":2},"source":"/Users/cs/Desktop/local/pya/app/public/wp-content/themes/pyapc/assets/scss/assets/scss/global/_mixins.scss","original":{"line":383,"column":-1},"name":null}
    at SourceMapGenerator_validateMapping [as _validateMapping] (/Users/cs/Desktop/local/pya/app/public/wp-content/themes/pyapc/node_modules/webpack-sources/node_modules/source-map/lib/source-map-generator.js:298:13)
```

## Expected result

[todo]: # (update version info!)

ruby sass 3.X.y
```css
.top-content {
	...
}
.top-content__container {
	...
}
```

http://libsass.ocbnet.ch/srcmap/#QG1peGluIHNxdWVlemUgewoKCUBtZWRpYSBzY3JlZW4gYW5kICggbWF4LXdpZHRoOiAxMjMwcHggKSB7CgkJcGFkZGluZy1sZWZ0OiA2MHB4OwoJCXBhZGRpbmctcmlnaHQ6IDYwcHg7Cgl9Cn0KCkBtaXhpbiB0b3AtY29udGVudCB7CgliYWNrZ3JvdW5kLWNvbG9yOiAjZWZlZmVmOwoJY29sb3I6ICNmZmY7CglwYWRkaW5nOiAxNXB4OwoJYm9yZGVyLXJhZGl1czogNHB4OwoKCSZfX2NvbnRhaW5lciB7CgoJCUBpbmNsdWRlIHNxdWVlemU7Cgl9Cn0KCi50b3AtY29udGVudCB7CgkKCUBpbmNsdWRlIHRvcC1jb250ZW50Owp9

version info:
```cmd
node            9.6.1
node-sass       4.7.2
libsass         3.5.0.beta.2
```

On Mac HS 10.13.3

Repo for testing can be found here: https://github.com/chancestrickland/pyapc (other Webpack/Node errors likely exist, but hung up on this one at the moment)

**Comment by xzyfer** [comment]
Please try the latest version of LibSass 3.5.2 available in node-sass 4.8.3

**Comment by chancestrickland** [comment]
Oi, sorry about that. Thanks!
